src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnectionOptions.cs

```diff
@@ -8,23 +8,54 @@
 using System.Net.WebSockets;
 using System.Security.Cryptography.X509Certificates;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Connections.Internal;
 
 namespace Microsoft.AspNetCore.Http.Connections.Client
 {
-    public class HttpOptions
+    public class HttpConnectionOptions
     {
+        private IDictionary<string, string> _headers;
+        private X509CertificateCollection _clientCertificates;
+        private CookieContainer _cookies;
+
+        public HttpConnectionOptions()
+        {
+            _headers = new Dictionary<string, string>();
+            _clientCertificates = new X509CertificateCollection();
+            _cookies = new CookieContainer();
+
+            Transports = HttpTransports.All;
+        }
+
         /// <summary>
         /// Gets or sets a delegate for wrapping or replacing the <see cref="HttpMessageHandlerFactory"/>
         /// that will make HTTP requests the server.
         /// </summary>
         public Func<HttpMessageHandler, HttpMessageHandler> HttpMessageHandlerFactory { get; set; }
 
-        public IReadOnlyCollection<KeyValuePair<string, string>> Headers { get; set; }
+        public IDictionary<string, string> Headers
```

Comment:
Auto-properties with initializers?

Reply:
It's not so as to throw if the collection is set to null.

Reply:
I don't think we usually do things that way with these. We generally throw at the consumer if the value is null.

Reply:
TBH though, I might prefer this. It's unambiguously wrong to set these to `null`, since we preinitialize them for you.